flow/Trace.cpp

```diff
@@ -229,33 +197,6 @@ struct TraceLog {
 		}
 	};
 
-	struct IssuesList : ITraceLogIssuesReporter, ThreadSafeReferenceCounted<IssuesList> {
```

Comment:
I am not sure I understand - why is that weird?

Reply:
It maybe just me but whenever I find the actual implementation of a declaration in a different file other than `XXX.cpp`, I feel uncomfortable.  😂

Reply:
OH I misread this change. Alright ignore my previous comments.

What I really want to say here is that `IssuesList` most likely would be used by other trace log writers in the future and thus keeping it in `Trace.h`(and its implementation in `Trace.cpp`) is probably a better idea.

Reply:
I am not sure I agree with that. The reason I took this out was because I was writing a new trace log writer and I didn't want to include `Trace.h`. Basically a simple file writer should not need to include all of `Trace.h` - smaller includes = fast compilation.

Reply:
I agree.

Reply:
I think the idea here is not necessarily that all log writers would need to use issues, but that some will and the idea should not be considered unique to the one implementation.

Also, if you are writing to a file, you may want to consider hooking up to the issues mechanism to ensure that errors working with the file get reported appropriately.

Reply:
yes, which is why the `IssuesList` is publicly accessible now.

This is just for preparation for a follow up PR. Basically this will eventually provide an additional mechanism to write traces (as opposed to log files). I want to reuse as much code as possible for this - but reusing the `IssuesList` implementation is hard if it is in a cpp file.

Reply:
To be more precise: the plan is to later provide an implementation of `ITracer` (can be found in `Tracing.h`) that will write to disk but to it's own file. Implementations that write to the network could also make use of the `ITraceLogIssuesReporter` but would probably provide their own implementation.